test/functional/api/autocmd_spec.lua

```diff
@@ -899,6 +899,89 @@ describe('autocmd api', function()
         eq([[:echo "Buffer"]], normalized_aus[1].command)
       end)
     end)
+
+    describe('id', function()
+      it('gets events by ID', function()
+        local id = api.nvim_create_autocmd('BufEnter', {
+          command = 'echo "hello"',
+        })
+        eq({
+          {
+            buflocal = false,
+            command = 'echo "hello"',
+            event = 'BufEnter',
+            id = id,
+            once = false,
+            pattern = '*',
+          },
+        }, api.nvim_get_autocmds({ id = id }))
+      end)
+
+      it('gets events by ID and group', function()
```

Comment:
let's just have 1 test for the "with other filters" case. we don't need to test every permutation. you can merge some of the filters from the below tests into this test, if possible, then drop the tests below.

```suggestion
      it('gets events by ID and other filters', function()
```